_posts/2016-11-14-JS-Glossary.md

```diff
@@ -0,0 +1,554 @@
+---
+layout: post
+title: "JavaScript Glossary for 2017"
+date: 2016-11-14 12:17
+author: orta
+categories: [javascript, emission, danger]
+series: React Native at Artsy
+---
+
+Getting to grips with the entire JavaScript ecosystem is a tough job when you're getting started. Coming from the native mobile space, there's a lot to learn. I've spent a lot of time in the environment now, and can distil so you can grok, then dig into places when you choose. This post is semi-opinionated, with links for further reading so you can get a different perspective too.
+
+This post focus specifically on the JavaScript tooling around React Native projects, but is applicable to all JavaScript projects. 
+
+<!-- more -->
+
+# React 
+
+### React
+
+React is a Facebook project which offers a uni-direction Component model that _can_ replace MVC in a front-end application. It was built out of a desire to mock a web page's view hierarchy (called the DOM) so that they could make changes as differences between view states.
+
+Its model is that you would create a set of Components to encapsulate each part for the state of the page. React makes it easy to make components that are functional in the FRP sense. They act like a function which takes some specially declared state and it is rendered into HTML.
+
+A component optionally uses a language called [JSX](#jsx) to visualise how each component's child components are set up,here's an example of a React component using JSX [from Emission, our React Native library][search-bar]:  
+
+```js
+export default class SearchBar extends React.Component {
+  render() {
+    return (
+      <TouchableWithoutFeedback onPress={this.handleTap.bind(this)}>
+        <View style={styles.container}>
+          <Image style={styles.searchIcon} source={require('../../../images/SearchButton.png')}/>
+          <Text style={styles.text}>Search for artists and artworks...</Text>
+        </View>
+      </TouchableWithoutFeedback>
+    )
+  }
+
+  handleTap() {
+    Switchboard.presentModalViewController(this, '/search')
+  }
+}
+```
+
+By providing a well encapsulated Component model, you can aggressively reduce the amount of redundant code you need to build an application. By not initially writing to the DOM, React can decide what has changed between user actions and that means you have to juggle significant less [state](#state).
+
+### React Native
+
+Writing native apps is now officially a pain. I came to this conclusion early this year, and it's been amazing to be able to work in React Native.
```

Comment:
Maybe notice that in the first sentence above that you are talking about tools that we available prior to `React native`? Otherwise it was not clear to me immediately that you are not talking about React Native but comparing it to the old tools.